golang/UPDATING.md

```diff
@@ -4,38 +4,21 @@
 
 #### One-Time Setup
 
-1. Download and setup Protocol Buffer release from https://github.com/protocolbuffers/protobuf/releases (if you haven't already done this for another language).  As of February 2019 we're using v3.7 release, which is compatible with proto2 .proto files.
-1. Download [Go](https://golang.org/dl/)
+1. Download and install [Docker](https://docs.docker.com/get-docker/)
 
 #### Every time `gtfs-realtime.proto` changes
 
-1. Regenerate the language binding source from gtfs-realtime.proto by running the following from the `golang` directory:
+1. Run the following from the project root folder:
 
     ```
-    go get -u github.com/golang/protobuf/protoc-gen-go
-    protoc --go_out=./gtfs/ --proto_path=.. ../gtfs-realtime.proto
+    docker build -t gtfs-golang -f golang/Dockerfile .
```

Comment:
Thanks for making the uni-test change.  Can you add an intermediate step here, something like "Verify that the build and tests ran successfully before proceeding to the next step."

As currently written, if you just copy/pasted the two docker commands, I believe you'd successfully run both even if that actual unit test failed.

Reply:
@bdferris-v2 I don't think that's actually the case! If the `build` doesn't succeed, then docker will never tag the image, which the subsequent `run` step references.

For example:
```
| $ docker build -t gtfs-golang -f golang/Dockerfile .
# ... omit for brevity...
 => ERROR [13/13] RUN go test ./...                                                                                                                                                                        4.2s
------
 > [13/13] RUN go test ./...:
#17 0.285 go: downloading google.golang.org/protobuf v1.26.0
#17 4.070 --- FAIL: TestFeedMessage (0.00s)
#17 4.070     gtfs_test.go:29: Expected entity length: 2, got: 1
#17 4.070 FAIL
#17 4.070 FAIL	github.com/MobilityData/gtfs-realtime-bindings/golang/gtfs/gtfs	0.003s
#17 4.070 FAIL
------
executor failed running [/bin/sh -c go test ./...]: exit code: 1
_________    | ~/gtfs-realtime-bindings
| $ docker run -it --rm gtfs-golang cat /lib/gtfs/gtfs-realtime.pb.go | tr -d '\r' > golang/gtfs/gtfs-realtime.pb.go
Unable to find image 'gtfs-golang:latest' locally
```

Reply:
Hmmm... I guess in my case, I still had a previous image tagged with gtfs-golang kicking around because I had already successfully built once and then artificially caused the test to fail for subsequent builds.  I guess theoretically, the subsequent `docker run -it --rm gtfs-golang` should be cleaning up the image, but do you think it's conceivable that the image might be around for other reasons?  I guess I want to avoid some condition where a test failed but you pull the gtfs-realtime.pb.go from a previously-built instance.  Maybe not that likely?

Reply:
@bdferris-v2 Ah, you bring up a good point! The previous image tag `gtfs-golang` was never cleaned up. `docker run --rm` was cleaning up the container that was executing the copy from the build to the host file, not the image. I added `docker rmi` at the beginning and turned the whole thing into a one-liner to avoid folks accidentally hitting weirdness with partial states.

I also refactored some of the `Dockerfile` and utilized `docker cp` to more-or-less push everything into docker to minimize potential variations in shell behavior